util/argo/resource_tracking.go

```diff
@@ -244,3 +245,20 @@ func (rt *resourceTracking) Normalize(config, live *unstructured.Unstructured, l
 
 	return nil
 }
+
+func (rt *resourceTracking) RemoveAppInstance(un *unstructured.Unstructured) error {
+	err := kube.RemoveAnnotation(un, common.AnnotationKeyAppInstance)
+	if err != nil {
+		return err
+	}
+	err = kube.RemoveAnnotation(un, common.AnnotationInstallationID)
+	if err != nil {
+		return err
+	}
+	err = kube.RemoveLabel(un, common.LabelKeyAppInstance)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Instead of removing all known tracking label/annotations, can we dynamically remove just the ones that we know were set by `SetAppInstance`? 

If someone is using annotation-based tracking and sets the `app.kubernetes.io/instance` label on a resource in git, we should only remove the annotation, not the label.